src/Provider/GoogleMaps/GoogleMaps.php

```diff
@@ -267,6 +269,19 @@ private function updateAddressComponent(AddressBuilder $builder, string $type, $
                 $builder->addAdminLevel(intval(substr($type, -1)), $values->long_name, $values->short_name);
                 break;
 
+            case 'sublocality_level_1':
+            case 'sublocality_level_2':
+            case 'sublocality_level_3':
+            case 'sublocality_level_4':
+            case 'sublocality_level_5':
+                $newSubLocalityLevel = [
+                    'level' => intval(substr($type, -1)),
+                    'name' => $values->long_name,
+                    'code' => $values->short_name,
+                ];
+                $builder->setValueInArray('subLocalityLevel', $newSubLocalityLevel);
```

Comment:
Im sorry, I dont know how to to that. 

This function `updateAddressComponent()` is called foreach `types`. That means it will be called multiple times for each `address_components`, and each time array `$newSubLocalityLevel ` will be reset, and it will always contains just one, current, `subLocalityLevel`. 

And function `setValue(..)` will always reassign `$data['subLocalityLevel']`, and set it to the current `subLocalityLevel`.

```
   foreach ($result->address_components as $component) {
       foreach ($component->types as $type) {
               $this->updateAddressComponent($builder, $type, $component);
        }
   }
```
and here is result

```
{
   "results" : [
      {
         "address_components" : [
            {
               "long_name" : "58",
               "short_name" : "58",
               "types" : [ "political", "sublocality", "sublocality_level_4" ]
            },
            {
               "long_name" : "Iijima",
               "short_name" : "Iijima",
               "types" : [ "political", "sublocality", "sublocality_level_1" ]
            }
  ]
}

```